File: beginner_tutorials/src/changeColor.py
#!/usr/bin/env python3
#18070006034 Nuri Can ÖZTÜRK
#chmod u+x ~/catkin_ws/src/beginner_tutorials/src/changeColor.py

import rospy
from std_srvs.srv import Empty # call'İn
from turtlesim.srv import *


rospy.init_node('colorr',anonymous=True)

color_r = rospy.get_param('r')
color_g = rospy.get_param('g')
color_b = rospy.get_param('b')



print(color_r,color_g,color_b)
rospy.wait_for_service("clear")  #active clear service

rospy.set_param("background_r",color_r)
rospy.set_param("background_g",color_g)
rospy.set_param("background_b",color_b)

clear_background = rospy.ServiceProxy("/clear",Empty) # service_name , service_type
clear_background()

File: beginner_tutorials/src/changeColor.cpp
#include <ros/ros.h>
#include <std_srvs/Empty.h>
#include <turtlesim/Spawn.h>

int main(int argc, char **argv)
{
    ros::init(argc,argv,"setcolor");
    ros::NodeHandle n;
    int r,g,b;
    n.getParam("r",r);
    n.getParam("g",g);
    n.getParam("b",b);

    ros::service::waitForService("clear");
    ros::param::set("background_r",r);
    ros::param::set("background_g",g);
    ros::param::set("background_b",b);

    ros::ServiceClient clearClient = n.serviceClient<std_srvs::Empty>("clear");
    std_srvs::Empty srv;
    clearClient.call(srv);
    return 0;
}